manimlib/mobject/svg/svg_mobject.py

```diff
@@ -29,136 +32,129 @@ class SVGMobject(VMobject):
         "should_center": True,
         "height": 2,
         "width": None,
-        # Must be filled in a subclass, or when called
         "file_name": None,
+        # Style that overrides the original svg
         "color": None,
         "opacity": None,
         "fill_color": None,
         "fill_opacity": None,
         "stroke_width": None,
         "stroke_color": None,
         "stroke_opacity": None,
-        "path_string_config": {}
+        # Style that fills only when not specified
+        # If None, regarded as default values from svg standard
+        "svg_default": {
+            "color": None,
+            "opacity": None,
+            "fill_color": None,
+            "fill_opacity": None,
+            "stroke_width": None,
+            "stroke_color": None,
+            "stroke_opacity": None,
+        },
+        "path_string_config": {},
     }
 
     def __init__(self, file_name=None, **kwargs):
-        digest_config(self, kwargs)
-        self.file_name = file_name or self.file_name
-        if file_name is None:
-            raise Exception("Must specify file for SVGMobject")
-        self.file_path = get_full_vector_image_path(file_name)
-
         super().__init__(**kwargs)
+        self.file_name = file_name or self.file_name
+        self.init_svg_mobject()
+        self.init_colors()
         self.move_into_position()
 
-    def move_into_position(self):
-        if self.should_center:
-            self.center()
-        if self.height is not None:
-            self.set_height(self.height)
-        if self.width is not None:
-            self.set_width(self.width)
-
-    def init_colors(self):
-        # Remove fill_color, fill_opacity,
-        # stroke_width, stroke_color, stroke_opacity
-        # as each submobject may have those values specified in svg file
-        self.set_stroke(background=self.draw_stroke_behind_fill)
-        self.set_gloss(self.gloss)
-        self.set_flat_stroke(self.flat_stroke)
-        return self
-
-    def init_points(self):
-        with open(self.file_path, "r") as svg_file:
-            svg_string = svg_file.read()
+    def init_svg_mobject(self):
+        hasher = hashlib.sha256(str(self.hash_seed).encode())
```

Comment:
There should be a better way to hash a tuple with dict, see https://github.com/3b1b/manim/pull/1744#discussion_r806411418

Reply:
Just updated. You may update #1744 using the new function defined in `manimlib/utils/iterables.py`.